src/Stratis.Bitcoin.Features.BlockStore/BlockStoreLoop.cs

```diff
@@ -16,7 +16,7 @@ namespace Stratis.Bitcoin.Features.BlockStore
     /// <summary>
     /// The BlockStoreLoop simultaneously finds and downloads blocks and stores them in the BlockRepository.
     /// </summary>
-    public class BlockStoreLoop
+    public class BlockStoreLoop : IStoreStateProvider
```

Comment:
Probably making an interface for the whole `BlockStoreLoop` would be a better idea but I'm fine with the quick approach since block store loop will be refactored soon anyway